Day2/ex1.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"fmt"
+	"sync"
+)
+
+type Frequency struct { // Struct for overall frequency
+	frequency map[string]int
+	mutex sync.Mutex
+}
+
+func (frequency *Frequency) FrequencyCalculator(id int, jobs <-chan string, wg *sync.WaitGroup)  { // go routine for calculating frequency of a word
+
+	defer wg.Done()
+	word := <- jobs
+
+	for _, letter := range word	{
+			frequency.mutex.Lock()
+			frequency.frequency[string(letter)] += 1
+			frequency.mutex.Unlock()
+	}
```

Comment:
numOfWorkers are coupled with numOfWords

if numOfWorkers < numOfWords then deadlock occurs at line 42
if numOfWorkers > numOfWords then subset of subroutines get blocked indefinitely

```suggestion
	
     for word := range jobs {
	for _, letter := range word	{
			frequency.mutex.Lock()
			frequency.frequency[string(letter)] += 1
			frequency.mutex.Unlock()
	}
}
```